modules/mono/glue/GodotSharp/GodotSharp/Core/Bridge/ScriptManagerBridge.types.cs

```diff
@@ -20,7 +20,10 @@ public void Add(IntPtr scriptPtr, Type scriptType)
             // TODO: What if this is called while unloading a load context, but after we already did cleanup in preparation for unloading?
 
             _scriptTypeMap.Add(scriptPtr, scriptType);
-            _typeScriptMap.Add(scriptType, scriptPtr);
+            
+            // Due to generic classes, more than one class can point to the same type, so
+            // there could be duplicate keys in this case. We only add a type as key once.
+            _typeScriptMap.TryAdd(scriptType, scriptPtr);
```

Comment:
You are right, filtering the types would be the best approach. I didn't find a way to do that. 😐

I you have any idea when the types are created, that would help.

Reply:
I think you need to update the other entry as well (removing the old script pointer). Based on my test the change I propose can reduce the issue into an Editor Error Message instead of Crashing the Mono module (Fail to unload assembly on second build). However, I agree that finding the root cause is necessary.

```csharp
public void Add(IntPtr scriptPtr, Type scriptType)
{
    // TODO: What if this is called while unloading a load context, but after we already did cleanup in preparation for unloading?

    if (_typeScriptMap.TryGetValue(scriptType, out var oldPtr))
    {
        _scriptTypeMap.Remove(oldPtr);
        _typeScriptMap[scriptType] = scriptPtr;
    }
    else
    {
        _typeScriptMap.Add(scriptType, scriptPtr);
    }
    _scriptTypeMap.Add(scriptPtr, scriptType);
```